src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java

```diff
@@ -94,20 +94,54 @@ boolean isDirty() {
 
     public void openFile(@Nullable Long offset) throws IOException {
         SourceFile fileProps = new SourceFile();
-
         File folder = new File(basePath);
         if (!folder.exists() && !folder.mkdirs()) {
             if (!folder.exists()) {
                 throw new IOException(String.format("Failed to create new directory %s", folder.getPath()));
             }
             log.warn("Couldn't create the directory because it already exists (likely a race condition)");
         }
-
         String filePath = getFilePath.apply(offset);
         fileProps.path = filePath;
-        File file = new File(filePath);
+        // Sanitize the file name just be sure and make sure it has the R/W permissions only
+        String sanitizedFilePath = FilenameUtils.normalize(filePath);
+        if (sanitizedFilePath == null) {
+            /*
+             * This condition should not occur at all. The files are created in controlled manner with the names consisting DB name, table name. These do not
+             * permit names like "../../" or "./" etc. Still adding an additional check.
+             */
+            String errorMessage = String.format("Exception creating local file for write." +

```

Comment:
lets keep one sort of formating - either %s or {}

Reply:
Sure, noted Ohad. in the logs it has been {} , here we created the message to throw. Hence used the format with %s. Hope that is okay